### ISSUE TITLE
_layernorm implementation forward result not equal F.layer_norm

### ISSUE THREAD

**Issue by Tengxu-Sun** [lucidrains/triton-transformer#6]
I have a try on your triton-transformer and test the layernorm module alone. It's very weird that the forward result is different  while the backward result is equal.

code:
from triton_transformer.layernorm import layernorm
import torch
import torch.nn as nn

torch.manual_seed(0)
x = torch.randn(2,5).cuda()
x.requires_grad_(True)
dy = .1*torch.randn_like(x).cuda()
dim = 5
norm = nn.LayerNorm(dim).cuda()

y1 = layernorm(x, norm.weight, norm.bias, use_triton = True)
y2 = layernorm(x, norm.weight, norm.bias, use_triton = False)
print(y1, y2)
print(torch.allclose(y1, y2))

y1.backward(dy, retain_graph=True)
dx_y1 = x.grad.clone()

x.grad = None

y2.backward(dy, retain_graph=True)
dx_y2 = x.grad.clone()
print(dx_y1, dx_y2)
print(torch.allclose(dx_y1, dx_y2))

result:
`tensor([[ 0.9492, -0.0021, -0.9797,  0.4449, -0.4123],
        [-0.7624,  0.4399,  0.7299, -0.3091, -0.0983]], device='cuda:0',
       grad_fn=<_layernormBackward>) tensor([[ 1.4217, -0.0031, -1.4674,  0.6663, -0.6175],
        [-1.4342,  0.8276,  1.3732, -0.5815, -0.1850]], device='cuda:0',
       grad_fn=<NativeLayerNormBackward>)
False

tensor([[-0.0706,  0.0288, -0.0813,  0.0446,  0.0785],
        [ 0.0218, -0.0152,  0.0141, -0.0522,  0.0315]], device='cuda:0') tensor([[-0.0706,  0.0288, -0.0813,  0.0446,  0.0785],
        [ 0.0218, -0.0152,  0.0141, -0.0522,  0.0315]], device='cuda:0')
True`

**Comment by lucidrains** [comment]
@Tengxu-Sun oh crap, yes, you caught a :bug: https://github.com/lucidrains/triton-transformer/commit/ed24e653c018cf4edb6f4c1c5d29f82cde7b4ecb thank you!
